nexus-repository-conan/src/main/java/org/sonatype/repository/conan/upgrade/ConanUpgrade_1_1.java

```diff
@@ -0,0 +1,156 @@
+  }
+
+  @Override
+  public void apply() {
+    if (hasSchemaClass(configDatabaseInstance, "repository") &&
+        hasSchemaClass(componentDatabaseInstance, ASSET_CLASS_NAME)) {
+      List<String> repositoryNames = findConanRepositoryNames();
+      updateAssetPath(repositoryNames);
+      removeAttributesFromConanManifest(repositoryNames);
+    }
+  }
+
+  private List<String> findConanRepositoryNames() {
+    try (ODatabaseDocumentTx oDatabaseDocumentTx = configDatabaseInstance.get().connect()) {
+      final List<ODocument> documents = oDatabaseDocumentTx.query(
+          new OSQLSynchQuery<ODocument>("select from repository where recipe_name in ['conan-proxy', 'conan-hosted']"));
+      return documents
+          .stream()
+          .map(entries -> (String) entries.field(P_REPOSITORY_NAME))
+          .collect(Collectors.toList());
+    }
+  }
+
+  private void removeAttributesFromConanManifest(final List<String> repositoryNames) {
+    DatabaseUpgradeSupport.withDatabaseAndClass(componentDatabaseInstance, ASSET_CLASS_NAME,
+        (db, type) -> repositoryNames.forEach(repositoryName -> {
+          OIndex<?> bucketIdx = db.getMetadata().getIndexManager().getIndex(I_REPOSITORY_NAME);
+          OIdentifiable bucket = (OIdentifiable) bucketIdx.get(repositoryName);
+
+          List<ODocument> conanManifests = db.query(new OSQLSynchQuery<ODocument>(
+                  "select from asset where bucket = ? and attributes.conan.asset_kind = 'CONAN_MANIFEST'"),
+              bucket.getIdentity());
+
+          conanManifests
+              .forEach(oDocument -> {
+
+                Map<String, Object> attributes = oDocument.field("attributes");
+                // remove all attributes, except asset_kind from conan "bucket"
+                attributes.put(ConanFormat.NAME, Collections.singletonMap(P_ASSET_KIND, AssetKind.CONAN_MANIFEST.name()));
+                oDocument.field("attributes", attributes);
+
+                oDocument.save();
+              });
+        }));
+  }
+
+  private void updateAssetPath(final List<String> repositoryNames) {
+    DatabaseUpgradeSupport.withDatabaseAndClass(componentDatabaseInstance, ASSET_CLASS_NAME,
+        (db, type) -> repositoryNames
+            .stream()
+            .flatMap(repositoryName -> {
+              OIndex<?> bucketIdx = db.getMetadata().getIndexManager().getIndex(I_REPOSITORY_NAME);
+              OIdentifiable bucket = (OIdentifiable) bucketIdx.get(repositoryName);
+              List<ODocument> assets =
+                  db.query(new OSQLSynchQuery<ODocument>("select from asset where bucket = ?"), bucket.getIdentity());
+              return assets.stream();
+            })
+            .forEach(oDocument -> {
+              String name = oDocument.field("name");
+              String nextName = null;
+
+              if (name.startsWith("/v1/conans/v1/conans/")) { // broken HOSTED
+                nextName = "conans/" + name.substring(21);
```

Comment:
might replace `21` with `"/v1/conans/v1/conans/".length()` (and similar in the other else ifs below, just to keep folks from trying to figure what the number relates to)

Reply:
not a bad idea.
https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/95/commits/9938f7c67cac959dbbf6292c9271250ee6812827